public/app/features/alerting/unified/integration/AlertRulesDrawerContent.tsx

```diff
@@ -27,12 +40,17 @@ export default function AlertRulesDrawerContent({ dashboardUid }: Props) {
     .flatMap((g) => g.rules)
     .filter((rule) => rule.annotations[Annotation.dashboardUID] === dashboardUid);
 
+  const loading = loadingRulesData;
```

Comment:
why do we need this new variable `loadingRulesData`, why don't we use directly loading from here?:
`const { loading } = useAsync(async () => {`

Reply:
Should we include loading state of `fetchDashboard` too?